src/test/java/org/reaktivity/nukleus/maven/plugin/internal/generated/Map16FW.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright 2016-2019 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.maven.plugin.internal.generated;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.agrona.BitUtil;
+import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.reaktivity.reaktor.internal.test.types.Flyweight;
+
+public final class Map16FW<KV extends VariantFW<?, ?>, VV extends VariantFW<?, ?>> extends MapFW<KV, VV>
+{
+    private static final int LENGTH_SIZE = BitUtil.SIZE_OF_SHORT;
+
+    private static final int FIELD_COUNT_SIZE = BitUtil.SIZE_OF_SHORT;
+
+    private static final int LENGTH_OFFSET = 0;
+
+    private static final int FIELD_COUNT_OFFSET = LENGTH_OFFSET + LENGTH_SIZE;
+
+    private static final int FIELDS_OFFSET = FIELD_COUNT_OFFSET + FIELD_COUNT_SIZE;
+
+    private static final int LENGTH_MAX_VALUE = 0xFFFF;
+
+    private final KV keyRO;
+
+    private final VV valueRO;
+
+    private final DirectBuffer entriesRO = new UnsafeBuffer(0L, 0);
+
+
+    public Map16FW(
+        KV keyRO,
+        VV valueRO)
+    {
+        this.keyRO = keyRO;
+        this.valueRO = valueRO;
+    }
+
+    @Override
+    public int length()
+    {
+        return buffer().getShort(offset() + LENGTH_OFFSET);
+    }
+
+    @Override
+    public int fieldCount()
+    {
+        return buffer().getShort(offset() + FIELD_COUNT_OFFSET);
+    }
+
+    @Override
+    public DirectBuffer entries()
+    {
+        return entriesRO;
+    }
+
+    @Override
+    public void forEach(
+        Function<KV, Consumer<VV>> consumer)
+    {
+        int offset = offset() + FIELDS_OFFSET;
+        for (int i = 0; i < fieldCount() / 2; i++)
```

Comment:
Hoist `fieldCount` to avoid re-executing per loop iteration, and consider `i += 2` instead of division.

Same feedback for other `MapNFW` classes.